src/Magnum/ImGuiIntegration/Context.cpp

```diff
@@ -48,12 +49,46 @@
 #include <Magnum/GL/TextureFormat.h>
 #include <Magnum/GL/Version.h>
 #include <Magnum/Math/Matrix3.h>
+#include <Magnum/Math/Range.h>
 
 #include "Magnum/ImGuiIntegration/Integration.h"
 #include "Magnum/ImGuiIntegration/Widgets.h"
 
+#define IMGUI_HAS_DYNAMIC_TEXTURES (IMGUI_VERSION_NUM >= 19200)
+
+#if IMGUI_HAS_DYNAMIC_TEXTURES
+#include <imgui_internal.h> /* GetPlatformIO(ImGuiContext*) */
+#endif
+
 namespace Magnum { namespace ImGuiIntegration {
 
+namespace {
+
+#if IMGUI_HAS_DYNAMIC_TEXTURES
+void updateTexture(ImTextureData& texture, const Range2Di& rect) {
+    const Vector2i offset = rect.min();
+    PixelStorage storage;
+    /* Data is tightly packed and rect can have any size */
+    storage.setAlignment(1);
+    storage.setRowLength(texture.Width);
+    storage.setSkip({offset.x(), offset.y(), 0});
```

Comment:
Is this safe on every platform, specifically WebGL1? The ImGui code does some manual packing into a tight buffer

Reply:
I copypasted what `GlyphCacheGL` does, kinda cleans up nicely when combined with only updating the combined rect